packages/flutter_tools/lib/src/compile.dart

```diff
@@ -17,6 +17,8 @@ import 'base/logger.dart';
 import 'base/platform.dart';
 import 'build_info.dart';
 import 'convert.dart';
+import 'globals.dart' as globals;
```

Comment:
please don't introduce more globals usage

Reply:
either inject a helper interface to do the plugin implementation, or provide the globally used interfaces directly

Reply:
Done